### PR TITLE
Add Geolocation API to get the user's location automatically

### LINKED ISSUE #129: Use browser Geolocation API to get the user's location automatically.
[Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API) on MDN Web Docs.

### Tasks
- [x] Add a `BIconGeoAltFill` icon to the `LocationPicker` component (to the right of the City input).
- [x] Create a handler when the icon is clicked.
- [x] Create `src/services/GeolocationService.js` with a `getCurrentPosition` function.
- [x] Call `GeolocationService.getCurrentPosition` in the `BIconGeoAltFill` click handler and save result as `coords`.
- [x] Set `$store.location.lat` to `coords.latitude` and $store.location.lon` to coords.longitude`
- [x] Handle errors with a notice to the user. Consider a [Toast](https://bootstrap-vue.org/docs/components/toast).
- [x] Set `$store.location.city` based on the lat/lon. Not sure how to do this.

### Example code for getCurrentPosition service function

Based on the [MDN example](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API#javascript).

```javascript
return new Promise((resolve, reject) => {
  if(!navigator.geolocation) {
    reject('Geolocation is not supported by your browser');
  }
  navigator.geolocation.getCurrentPosition(() => resolve(position.coords), () => reject('Unable to retrieve your location');
});
```

### PR DESCRIPTION
**Pull request by mjp2233**
This PR Closes #129 

## Checklist before requesting approval

- [x] All PR checks succeed
- [x] This branch does not have any conflicts with the master branch.
- [x] The version number has been incremented appropriately (we use [Semver](https://semver.org/) as a versioning strategy).

### CODE REVIEW

**Review by dave-burke — COMMENTED** [review]
Looks like a great start. I like the use of `b-input-group-append` for the icon. It looks nice! Hope this helps. I'll try to keep an eye out on my email today if you have any more questions.

File: src/services/GeoLocationService.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    // const resolve = (position: { coords: { longitude: any; latitude: any; }; }): void => {
+    //   store.commit(SET_LON, position.coords.longitude);
+    //   store.commit(SET_LAT, position.coords.latitude);
+    // };
+
+    return new Promise((resolve, reject) => {
+      if (!navigator.geolocation) {
+        reject(new Error('Geolocation is not supported by your browser'));
+      }
+      navigator.geolocation.getCurrentPosition(() => resolve(position.coords), () => reject(new Error('Unable to retrieve your location')));
```
[the remark this answers is not in the document]
Comment (mjp2233) [reply src/services/GeoLocationService.ts]: Good catch, that worked

**Comment by mjp2233** [comment]
@dave-burke Thanks for the help, I got it working and was able to console log the Lat/Long in the locationPicker.

The next thing I was going to think about was getting the City from Coordinates. I'm thinking we could leverage the OpenWeatherMap API since it takes in coordinates and returns a city along with the weather.

File: src/services/GeoLocationService.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import openWeather from '@/services/openWeatherMap';
+import { SET_CITY, SET_LAT, SET_LON } from '../store/mutations';
+
+export default class GeoLocationService {
+  static errorMessge: string = 'Unable to retrieve your location';
+
+  private output: object = {};
+
+  static getCurrentPosition(): Promise<any> {
+    // Possible TODO Create object for coords instead of using 'any'
```
Comment (mjp2233) [inline src/services/GeoLocationService.ts:11]: I ended up setting the return Promise type to 'any' so I wouldn't get errors trying to get the lat/long off of the object. I was getting "Property does not exist on Object" when using Promise<object>. I'm thinking there is a way to define the object specifying the properties but haven't looked into it

Comment (dave-burke) [reply src/services/GeoLocationService.ts:11]: I wondered about this. @bburggraff do you know if there are existing type definitions for standard browser APIs? Could we just import the types for `navigator.geolocation` somehow?

File: src/components/BottomBar/LocationPicker.vue [hunk 3]
```diff
@@ -28,6 +36,32 @@ export default class LocationPicker extends Vue {
...
+    await geolocationService.getCurrentPosition().then(async (coords) => {
+      this.$store.commit(SET_LAT, coords.latitude);
+      this.$store.commit(SET_LON, coords.longitude);
+      // This might not be the best approach to displaying the City name and
+      // the converstion from coords to city might be alittle off
+      // TODO: We could remove the city setting all together and blank out
+      // the city since they are searching by coordinates and may not be in a city
+      const location: GeoDirectResponse[] = await openWeatherService
+        .searchCityByCoords(coords.latitude, coords.longitude);
+      this.$store.commit(SET_CITY, { city: location[0].name });
```
Comment (mjp2233) [inline src/components/BottomBar/LocationPicker.vue:49]: I wasn't sure on this so I left as a comment for future discussion.

File: src/components/BottomBar/LocationPicker.vue [hunk 4]
```diff
@@ -28,6 +36,32 @@ export default class LocationPicker extends Vue {
...
+        .getOneCallWeather(coords.latitude, coords.longitude);
+      this.$store.commit(SET_WEATHER, weather);
+    }).catch((error) => {
+      this.$bvToast.toast(`I'm sorry, there seems to be an issue. Error Message: ${error.message}`,
+        {
+          title: 'Oopsy-Daisy',
+          variant: 'danger',
+          solid: true,
+          toaster: 'b-toaster-top-center',
+        });
```
Comment (mjp2233) [inline src/components/BottomBar/LocationPicker.vue:61]: I tried to think of a Minnesota error message, maybe that'll add to our "secret sauce"

Comment (dave-burke) [reply src/components/BottomBar/LocationPicker.vue:61]: Haha, I love it! BTW, somehow I missed the notification that this was ready for review. Sorry it sat so long! I'll take a look at the rest of it this weekend.

**Comment by dave-burke** [comment]
I'm just trying to test this, and tweaking the error messages to handle errors I'm getting. It's nothing to do with the code, which is otherwise solid. It's things like: my laptop doesn't support geolocation. My phone wouldn't allow access unless the page was served over https (today I learned [you can configure the dev server to use a self-signed cert](https://forum.vuejs.org/t/vue-cli-serve-with-https/39359)). Then I made the mistake of deleting and regenerating my API key, so my new one isn't activated yet. I'll try again later today.

**Review by dave-burke — APPROVED** [review]
Everything works for me now! It's really cool to see this working. Great work!

**Review by dave-burke — APPROVED** [review]

**Comment by mjp2233** [comment]
Awesome, thanks for adding all the error handling. That will definitely be useful if we run into errors. I agree, it's really coming together!

**Pull request merged**